modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImport/ModalImportContent.tsx

```diff
@@ -195,11 +212,31 @@ export function ModalImportContent({
 									try {
 										const JSONFile = JSON.parse(
 											fileReader.result as string
-										) as {externalReferenceCode: string};
-										setError(undefined);
-										setExternalReferenceCode(
-											JSONFile.externalReferenceCode
-										);
+										) as
+											| {externalReferenceCode: string}
+											| ObjectDefinition[];
+
+										if (
+											Liferay.FeatureFlags[
+												'LPS-187142'
+											] &&
+											Array.isArray(JSONFile) &&
+											JSONFile[0]
+												.objectFolderExternalReferenceCode
+										) {
```

Comment:
Hey @danvb112, if the user try to import a JSON with objects without Folders. Like the old versions of object definitions. 

This condition will never be true:
`JSONFile[0].objectFolderExternalReferenceCode`

Has this scenario been mapped?

Reply:
Hey @igor-franca, as we discussed, I put `scope` instead of `objectFolderExternalReferenceCode` as it is the only property that is unique and required by an `objectDefinition`